pykern/pkio.py

```diff
@@ -225,16 +225,20 @@ def save_chdir(dirname, mkdir=False, is_pkunit_prefix=False):
             pkunit_prefix = prev_ppp
 
 
-def sorted_glob(path):
+def sorted_glob(path, key=None):
     """sorted list of py.path.Local objects, non-recursive
 
     Args:
         path (py.path.Local or str): pattern
+        key (str): name of a function of path used to sort
 
     Returns:
         list: py.path.Local objects
     """
-    return sorted(py_path(f) for f in glob.glob(str(path)))
+    return sorted(
+        (py_path(f) for f in glob.glob(str(path))),
+        key=(lambda p: getattr(p, key)()) if key else None,
```

Comment:
Good. I almost thought it would be better to remove the invariant, but that would be reaching inside the py.path abstraction so better this way. :)